src/main/java/com/sunnylabs/tracegenerator/Configuration.java

```diff
@@ -0,0 +1,282 @@
+
+                o.addCall(op2);
+            }
+        });
+
+        // cross app calls
+        for (int i = 0; i < apps.size(); i++) {
+            Operation random = getRandom(getRandom(getRandom(apps).getServices()).getOperations());
+            Operation target = getRandom(getRandom(getRandom(apps).getServices()).getOperations());
+            while (target.getApplication().equals(random.getApplication())) {
+                target = getRandom(getRandom(getRandom(apps).getServices()).getOperations());
+            }
+            random.addCall(target);
+        }
+    }
+
+    private <T> T getRandom(Map<?, T> from) {
+        return new ArrayList<>(from.values()).get(new Random().nextInt(from.size()));
+    }
+
+    private <T> T getRandom(Collection<T> from) {
+        return new ArrayList<>(from).get(new Random().nextInt(from.size()));
+    }
+
+
+    private Application createRandomApp(List<String> serviceNames, List<String> operationNames) {
+        Application a = new Application();
+        Map<String, Service> services = new HashMap<>();
+        int desiredServices = Math.min(servicesPerApp, serviceNames.size());
+        for (int i = 0; i < desiredServices; i++) {
+            int idx = new Random().nextInt(serviceNames.size());
+            String name = serviceNames.get(idx);
+            serviceNames.remove(idx);
+            Service service = createRandomService(new ArrayList<>(operationNames));
+            services.put(name, service);
+        }
+        a.setServices(services);
+        return a;
+    }
+
+    private Service createRandomService(List<String> operationNames) {
+        Service s = new Service();
+        List<Operation> operations = new ArrayList<>();
+        for (int i = 0; i < 10; i++) {
+            int idx = new Random().nextInt(operationNames.size());
+            String name = operationNames.get(idx);
+            operationNames.remove(idx);
+            operations.add(new Operation(name));
+        }
+        s.setOperations(operations);
+        return s;
+    }
+
+    private void fixOperationReferences(List<Operation> calls) {
+        for (int i = 0; i < calls.size(); i++) {
+            final int index = i;
+            existingOperation(calls.get(index)).ifPresent(e -> calls.set(index, e));
+        }
+    }
+
+    private Optional<Operation> existingOperation(Operation c) {
+        Application app = getApplication(c.getApplication());
+        if (app == null) {
+            return Optional.empty();
+        }
+        Service svc = app.getService(c.getService());
+        if (svc == null) {
+            return Optional.empty();
+        }
+        return Optional.of(getOperationOrRandom(c, svc));
+    }
+
+    private Operation getOperationOrRandom(Operation c, Service svc) {
+        List<Operation> operations = svc.getOperations();
+        Optional<Operation> first = operations.stream().
+                filter(o -> o.getName().equals(c.getName())).findFirst();
```

Comment:
i see that Service.operations is a list while Application.services is a map.  Any particular reason for the difference?  A map would simplify this line

Reply:
Yep, I started with lists and switched to maps, but missed service.operations - fixed in e960d98

Reply:
i think `exampleConfig.yaml` should be updated too?